File: scripts/inspect_excel.py
import pandas as pd
import sys

try:
    # Load the Excel file
    file_path = 'references/Tabel Kriteria Evaluasi AKIP MA RI.xlsx'
    df = pd.read_excel(file_path, header=None)
    
    # Print the first 20 rows
    print("First 20 rows of the Excel file:")
    print(df.head(20).to_string())
    
    # Also verify the other file
    file_path_skor = 'references/Tabel Skor Penilaian Evaluasi AKIP.xlsx'
    df_skor = pd.read_excel(file_path_skor, header=None)
    print("\nFirst 20 rows of Skor file:")
    print(df_skor.head(20).to_string())

except Exception as e:
    print(f"Error: {e}")

File: scripts/inspect_excel.js
const XLSX = require('xlsx');
const fs = require('fs');
const path = require('path');

try {
    const filePath = path.join(process.cwd(), 'references', 'Tabel Kriteria Evaluasi AKIP MA RI.xlsx');
    const outputPath = path.join(process.cwd(), 'temp_inspection.txt');

    let output = `Reading file: ${filePath}\n`;

    if (!fs.existsSync(filePath)) {
        output += 'File not found!\n';
        fs.writeFileSync(outputPath, output);
        process.exit(1);
    }

    const workbook = XLSX.readFile(filePath);
    const sheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[sheetName];

    const data = XLSX.utils.sheet_to_json(worksheet, { header: 1 }); // Array of arrays

    output += 'First 50 rows of Criteria (Sheet 1):\n';
    data.slice(0, 50).forEach((row, i) => {
        output += `Row ${i}: ${JSON.stringify(row)}\n`;
    });

    // Check second file
    const filePathSkor = path.join(process.cwd(), 'references', 'Tabel Skor Penilaian Evaluasi AKIP.xlsx');
    if (fs.existsSync(filePathSkor)) {
        const wb2 = XLSX.readFile(filePathSkor);
        const ws2 = wb2.Sheets[wb2.SheetNames[0]];
        const data2 = XLSX.utils.sheet_to_json(ws2, { header: 1 });
        output += '\nFirst 20 rows of Skor File:\n';
        data2.slice(0, 20).forEach((row, i) => {
            output += `Row ${i}: ${JSON.stringify(row)}\n`;
        });
    }

    fs.writeFileSync(outputPath, output);
    console.log('Output written to temp_inspection.txt');

} catch (error) {
    console.error('Error:', error);
}
